docs/cli-usage.md

```diff
@@ -0,0 +1,96 @@
+## Using Kubernetes from the command line
```

Comment:
Object names should be able to be specified on create and update command lines, also: `--name=foo`. The name field is the same in all objects. This would facilitate config reuse. An enscope-like `--name-suffix=` option would be useful, too.

Note that as much intelligent intra-object defaulting should be done in the apiserver as possible. Multi-object defaulting makes sense in kubectl and/or generators, but only if it has bounded exposure to the api surface -- as discussed in the [google-containers thread](https://groups.google.com/forum/#!topic/google-containers/lZcoeLAKYnY), we don't want to get into the business of needing to update kubectl every time we add a field to the API. I think command-line options to derive objects from related objects (create pod --from-pod foo, create rc --from-pod foo, create service --from-rc foo) meet this requirement, and expose could be handled this way, at least when combined with general mechanisms for specifying fields on the command line. I also think it's fine to surface the primary (most commonly used, stable) fields/subobjects on the command line (e.g., --container=`{image:"redis"}`), provided it's possible to merge those subparts of the object with ~arbitrary json for the rest.

On config generators:

We should add a new command, say `expand`, which just translates yaml to json, invokes generators, performs other transformations, populates other fields set by kubectl, such as namespace, apiVersion, kind, labels, annotations, etc., without sending the objects to the apiserver. 

Config generators should otherwise be invoked similarly to real API objects: get, create, update, delete. The generators would be invoked prior to any of the API calls in order to expand to the set of API objects. Generators should support the standard object metadata, kind, apiVersion, name, labels, annotations. Other than for kind, the other fields will be propagated to all objects created. Not all apiVersions might be supported by a particular generator.

We'll need a type registry in kubectl to distinguish generator plugins from API objects. 

Like the API objects, generators themselves will be versioned (say, `genVersion`), and kubectl will assume a default version for each generator. Other versions should be accessible either by setting the field in the JSON or via command-line flag.

Next I'll hash out run more specifically.